### DISCUSSION TITLE
How to check the url of the crazyflie?

### DISCUSSION THREAD

**Discussion by cc299792458** [bitcraze/discussions#660] [Q&A] [answered]
Hi,

I forget my crazyflie's url, which makes me failed to connect to it by cfclient. I tried 0xE7E7E7E7E0 to 0xE7E7E7E7E7, but it seemed none of them matched.

I also tried to connect crazyflie with my iphone's app, it sccussed but I can't check the url.

How can I check the url of my crazyflie? Thanks!

**Comment by knmcguire** [comment, marked as answer]
hi!

Easiest is a micro usb cable, use the cfclient and press scan, then connect to usb://0. Then you can go to the menu item 'connect' 'configure 2.X' and there you can see channel, speed and the address of the connect crazyflie.

If you haven't configured the usb permissions you'd need to do that for the crazyflie as well. Assuming  that you are using linux this are the instructions: https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/installation/usb_permissions/

**Comment by cc299792458** [comment]
Hi,

Thanks for the instruction! I tried to use a usb cable to connect the crazyflie and my pc(linux system). The left-behind led blinking in blue. However, after I press scan, the connect button is still grey which cannot press. 

I follow the usb permissions instructions and it says "group 'plugdev' already exists". I think maybe I have set that before.

Is there anything wrong? I attach a video for you reference. (The right-front led blinking in green not red because I modified the firmware).

https://user-images.githubusercontent.com/52767591/231971289-0c407397-ffd7-4136-96a6-af85be92828a.MOV

**Reply by ataffanel** [reply]
The LED pattern looks good: your Crazyflie has passed self-test and is currently charging.

You might need to logout and login again or to restart your computer for the USB permission to take effect.

When clicking scan, there should be a `usb://0` device in the list:
![image](https://user-images.githubusercontent.com/6381499/231979295-f3750ebe-bf39-41de-a743-824aa2168e19.png)

The Crazyflie USB ids is currently 0483:5740 so, to check that is it well connected and seen by your PC, you should be able to see it with `lsusb` on the console:
![image](https://user-images.githubusercontent.com/6381499/231980325-aab1e07f-304f-43d4-8b5c-fbecf3e4fb4d.png)

**Reply by cc299792458** [reply]
Thanks! I will try it and get back to you!

**Reply by cc299792458** [reply]
Hi! I check it by lsusb.
I found that I can see Crazyradio PA USB Dongle but I can't see Crazyflie 2.X. How can I fix that? 
![Screenshot from 2023-04-14 18-28-45](https://user-images.githubusercontent.com/52767591/232020547-742bc7c1-f433-41b7-9845-e3bbcf61405d.png)

**Reply by cc299792458** [reply]
Hi, I also found that when I start the cfclient, it reports "WARNING:cflib.crtp.usbdriver:Exception while scanning for Crazyflie USB: 'NoneType' object has no attribute 'bcdDevice' in the console. Is that a problem?

**Reply by ataffanel** [reply]
The radio warning should not be a problem.

The Crazyflie not showing up on `lsusb` is a major problem though, this is likely the cause. Make sure your USB cable is one with proper data lanes, maybe try to use another USB cable. It will unfortunately not show as Crazyflie, but it will have the vid:pid of `0483:5740`.

**Reply by cc299792458** [reply]
Hi, I try another cable and unfortunately it still does not work. Is there any requirements for the USB cable, such as USB 2.0 or USB 3.0? 

I guess I can see Crazyradio PA USB Dongle means I follow the instructions right.(https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/installation/usb_permissions/) So I agree with you that maybe the reason lies in hardware like usb cable.

**Reply by knmcguire** [reply]
Are you sure that you copied all the udev rules? as you can see here, there is also 'crazyflie over usb' https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/installation/usb_permissions/

Perhaps a restart is necessary as well.

**Reply by cc299792458** [reply]
Hi, thanks for all your kindly reply.

I check the file "99-bitcraze.rules" and I think 'crazyflie over usb' exists. But there are also 2 addition files called "99-crazyflie.rules" and "99-crazyradio.rules". Should I delete them? 

![6a416fa229f193ac1ae1bbe7bec9685](https://user-images.githubusercontent.com/52767591/232512933-d76ecc4e-8148-4dd3-bc9c-dad643135d6d.png)
![68edc5f985a99a04dc958324111dcd0](https://user-images.githubusercontent.com/52767591/232512945-e058dd67-fb93-4882-a387-383e344218a3.png)
![1604fe2f52576f92ad761fa192a6d53](https://user-images.githubusercontent.com/52767591/232512960-e73111e2-7ac4-4463-ae7a-ba9f55a0e578.png)

**Reply by knmcguire** [reply]
Ah yes that is probably not helping indeed. You can remove the overlapping rule files

Since you selected the answer as working, does that mean it helped?

**Reply by cc299792458** [reply]
Hi, I think the answer below from ataffanel is work for me while I think your method is easier and commoner as you said. I mark it as the answer because I hope the peers can find it first.

**Comment by cc299792458** [comment]
By the way, is there another method to read the uri without a usb cable?

**Reply by ataffanel** [reply]
This is not implemented, no, unfortunately.

It is possible to compile a firmware that does not set the address, this should start the Crazyflie with the default address whatever the configuration says and will allow you to change the configuration the same way you could have done it over USB. The only thing required would be to comment out [line 101 in radiolink.c](https://github.com/bitcraze/crazyflie-firmware/blob/9956e68e10ba52855634c08abde62fb093707814/src/hal/src/radiolink.c#L101). Then compile and flash the firmware using the radio bootloader.

**Reply by cc299792458** [reply]
Thank you so much for all your patient help. I think that would work.
